src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -463,6 +470,315 @@ int RGWBucket::check_object_index(const DoutPrefixProvider *dpp,
   return 0;
 }
 
+int RGWBucket::check_index_olh(const DoutPrefixProvider *dpp, RGWBucketAdminOpState& op_state,
+                                     RGWFormatterFlusher& flusher, const int shard, 
+                                     uint64_t* const count_out, optional_yield y) {
```

Comment:
could this per-shard coroutine just be a free function like `static int check_index_shard_olh()`? same for the per-shard version of `check_index_unlinked()`

Reply:
If I pass in the RadosStore and Bucket as args I could change it to a free function. It seems cleaner to me like this, with fewer function args, but I'm willing to change it if you prefer it that way?

Reply:
went ahead and made it a free function